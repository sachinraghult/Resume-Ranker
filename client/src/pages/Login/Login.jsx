import React, { useState, useContext } from "react";
import { Link, useNavigate } from "react-router-dom";
import axios from "axios";
import { Context } from "../../context/Context";
import { LoginStart, LoginSuccess, LoginFailure } from "../../context/Actions";
import Google from "../../components/GoogleSignLogin/Googlereg";
import styles from "./Login.module.css";

import { Url } from "../../enums";

import { SERVER_URL } from "../../config";

const Login = () => {
  const { dispatch } = useContext(Context);
  const navigate = useNavigate();

  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState(false);

  const handleSubmit = async (e) => {
    e.preventDefault();
    dispatch(LoginStart());
    setError(false);

    try {
      const res = await axios.post(`${SERVER_URL}/auth/login`, {
        email,
        password,
      });

      dispatch(
        LoginSuccess({
          ...res.data,
          type: res.data.type,
          authToken: res.data.token,
        })
      );
      navigate(Url.JOB_FEED, { replace: true });
    } catch (err) {
      dispatch(LoginFailure());
      setError(true);
    }
  };

  return (
    <div className={styles.loginWrapper}>
      <div className={styles.loginLeft}>
        <h3 className={styles.loginLogo}>Job Finder</h3>
        <span className={styles.loginDesc}>Enter your details...</span>
      </div>
      <div className={styles.loginRight}>
        <form className={styles.loginBox} onSubmit={handleSubmit}>
          <div className={styles.formtop}>
            <h2>Login</h2>
          </div>
          {/* <button className="googleButton">
            <img
              width="25px"
              alt="Google sign-in"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/53/Google_%22G%22_Logo.svg/512px-Google_%22G%22_Logo.svg.png"
            />
            Login with Google
          </button> */}
          <Google />
          <div className={styles.orContainer}>
            <div className={styles.line}>
              <span>
                ---------------------- or Login with Email
                ----------------------
              </span>
            </div>
          </div>

          <input
            type="email"
            placeholder="Email"
            className={styles.loginInput}
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            required
          />
          <input
            type="password"
            placeholder="Password"
            className={styles.loginInput}
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            required
          />
          {/* <span className={styles.Forgot}>
            <u>Forgot Password ?</u>
          </span> */}
          {error && (
            <span className={styles.loginError}>Invalid Credentials</span>
          )}
          <button
            data-testid="handleSubmit"
            className={styles.loginButton}
            type="submit"
          >
            Login
          </button>
          <span className={styles.caa}>
            Not Registered? <Link to={Url.SELECTION}>Create an Account</Link>{" "}
          </span>
        </form>
      </div>
    </div>
  );
};

export default Login;
