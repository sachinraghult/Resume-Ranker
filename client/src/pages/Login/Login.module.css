/* .Login {
    width: 100%;
    height: 100vh;
    background-color: #ffff;
    display: flex;
    justify-content: center;
    align-items: center;
    font-family: 'Roboto Slab', serif;
} */

/* @media screen and (max-width: 900px) {
    .Login{
        width: 100%;
        height: 100vh;
        background-color: #ffff;
        display: flex;
        justify-content: center;
        align-items: center;
        font-family: 'Roboto Slab', serif;
        flex-direction: column;
        background-color:red;
    }
} */

.loginError {
  color: red;
  font-size: 1.1rem;
  font-style: italic;
  display: flex;
  align-items: flex-start;
}

.loginWrapper {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 0%;
  font-family: "Roboto Slab", serif;
}

@media screen and (max-width: 900px) {
  .loginWrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    /* background-color: blue;  */
  }
}

.loginLeft {
  /* background-color: aqua;  */
  display: flex;
  padding-top: 10px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  margin: 0px;
  height: 100vh;
  background-image: url("../../../public/assests/img1.svg");
  background-repeat: no-repeat;
  object-fit: fill;
}

.loginRight {
  /* background-color: aqua; */
  width: 100%;
  display: flex;
  margin-left: 0px;
  flex-direction: column;
  /* justify-content: center; */
  align-items: center;
  /* background-color: aqua;  */
}

.orContainer {
  margin-top: 15px;
  margin-bottom: 10px;
}
.loginLogo {
  font-size: 50px;
  font-weight: 800;
  color: rgb(0, 83, 42);
}

.loginBox {
  height: 550px;
  margin-right: 100px;
  margin-top: 30px;
  padding-top: 0%;
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  /* background-color: aqua; */
}

@media screen and (max-width: 900px) {
  .loginBox {
    margin: auto;
    margin-top: 20px;
  }
  .loginLeft {
    height: 30vh;
  }
}
@media screen and (max-width: 330px) {
  .loginBox {
    margin: 0px;
    margin-top: 20px;
    padding: 0%;
  }
}

/* .loginBox span{
    text-align: center;
    margin-top: 10px;
} */

.loginBox label {
  font-size: 20px;
  margin-top: 15px;
}

.loginInput {
  width: 100%;
  height: 45px;
  border-radius: 4px;
  border: 1px solid gray;
  font-size: 18px;
  padding-left: 20px;
}

.loginInput:focus {
  outline: none;
}

.loginButton {
  margin-top: 5px;
  width: 100%;
  height: 50px;
  border-radius: 4px;
  border: 1px solid gray;
  font-size: 20px;
  background-color: rgb(0, 0, 0);
  color: white;
  cursor: pointer;
  margin-top: 20px;
}
.loginButton:hover {
  background-color: blue;
}
.googleButton {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 50px;
  border-radius: 4px;
  border: 1px solid gray;
  font-size: 20px;
  background-color: #fff;
  color: black;
  cursor: pointer;
  margin-top: 0%;
}

.googleButton img {
  margin-right: 20px;
}

.loginForgot {
  text-align: center;
  color: blue;
}

.loginRegisterButton {
  width: 80%;
  height: 40px;
  border-radius: 10px;
  border: 1px solid gray;
  font-size: 15px;
  padding: 0 20px;
  background-color: rgb(39, 83, 0);
  color: white;
  cursor: pointer;
}

.formtop {
  font-size: 25px;
  margin-bottom: 0%;
}

@media screen and (max-width: 330px) {
  .loginButton {
    width: 100%;
    height: 50px;
  }
  .googleButton {
    width: 100%;
    height: 50px;
  }
}

.Forgot {
  display: flex;
  justify-content: right;
  text-align: right;
  margin-bottom: 0%;
}
