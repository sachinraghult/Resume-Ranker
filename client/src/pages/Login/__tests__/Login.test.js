import React from "react";
global.React = React;
import ReactDOM from "react-dom/client";
import Login from "../Login";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";

it("renders without crashing", async () => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act(async () =>
    render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    )
  );
});

it("renders correctly", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Login />
    </BrowserRouter>
  );
  fireEvent.submit(getByTestId("handleSubmit"));
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
