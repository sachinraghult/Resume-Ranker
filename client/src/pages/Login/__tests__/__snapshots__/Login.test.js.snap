// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="loginWrapper"
>
  <div
    className="loginLeft"
  >
    <h3
      className="loginLogo"
    >
      Job Finder
    </h3>
    <span
      className="loginDesc"
    >
      Enter your details...
    </span>
  </div>
  <div
    className="loginRight"
  >
    <form
      className="loginBox"
      onSubmit={[Function]}
    >
      <div
        className="formtop"
      >
        <h2>
          Login
        </h2>
      </div>
      <div
        id="signInDiv"
      >
        Sign up
      </div>
      <div
        className="orContainer"
      >
        <div
          className="line"
        >
          <span>
            ---------------------- or Login with Email ----------------------
          </span>
        </div>
      </div>
      <input
        className="loginInput"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <input
        className="loginInput"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <button
        className="loginButton"
        data-testid="handleSubmit"
        type="submit"
      >
        Login
      </button>
      <span
        className="caa"
      >
        Not Registered? 
        <a
          href="/choose"
          onClick={[Function]}
        >
          Create an Account
        </a>
         
      </span>
    </form>
  </div>
</div>
`;
